### PR TITLE
MINOR: Update documentation for enabling optimizations

### PR DESCRIPTION
**Pull request by bbejeck**
Updated docs for enabling all optimizations as of `2.3`

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by bbejeck** [comment]
ping @guozhangwang, @mjsax, @vvcephei, @ableegoldman, @cadonna, @abbccdda

**Comment by bbejeck** [comment]
\cc @JimGalasyn @londoncalling

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -80,6 +80,14 @@ <h3><a id="streams_api_changes_230" href="#streams_api_changes_230">Streams API
         The new public interfaces <code>inMemoryWindowStore()</code> and <code>inMemorySessionStore()</code> are added to <code>Stores</code> and provide the built-in in-memory window or session store.
     </p>
 
+    <p>
+        As of 2.3.0 we've updated how to turn on optimizations. Now to enable optimizations, you need to do two things.
+        First, add this line to your properties <code>properties.setProperty(StreamsConfig.TOPOLOGY_OPTIMIZATION, StreamsConfig.OPTIMIZE);</code>, as you have done before.
```
Comment (londoncalling) [inline docs/streams/upgrade-guide.html]:
```suggestion
        First add this line to your properties <code>properties.setProperty(StreamsConfig.TOPOLOGY_OPTIMIZATION, StreamsConfig.OPTIMIZE);</code>, as you have done before.
```

File: docs/streams/upgrade-guide.html [hunk 2]
```diff
@@ -80,6 +80,14 @@ <h3><a id="streams_api_changes_230" href="#streams_api_changes_230">Streams API
...
+        Second, when constructing your <code>KafkaStreams</code> instance, you'll need to pass your configuration properties when building your
+        topology via the overloaded <code>StreamsBuilder.build(Properties)</code> method.
```
Comment (londoncalling) [inline docs/streams/upgrade-guide.html]:
```suggestion
        topology by using the overloaded <code>StreamsBuilder.build(Properties)</code> method.
```

**Review by londoncalling — APPROVED** [review]
- LGTM, with exception of suggestions (same as @JimGalasyn 's on the https://github.com/apache/kafka-site/pull/222)
-  Not sure what's going on with the JDK, Scala build problems

**Review by mjsax — APPROVED** [review]

**Comment by bbejeck** [comment]
cherry-picked to 2.3

**Pull request merged**